### PR TITLE
aws-okta 0.19.0 (new formula)

### PR DESCRIPTION
**Pull request by josacar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/aws-okta.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class AwsOkta < Formula
+  desc "Aws-vault like tool for Okta authentication"
```
Comment (fxcoudert) [inline Formula/aws-okta.rb]: I think `authenticate with AWS using your Okta credentials` is a better description

File: Formula/aws-okta.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  homepage "https://github.com/segmentio/aws-okta"
+  url "https://github.com/segmentio/aws-okta/archive/v0.19.0.tar.gz"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = p buildpath
```
Comment (fxcoudert) [inline Formula/aws-okta.rb]: That `p` here is weird

File: Formula/aws-okta.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    (buildpath/"src/github.com/segmentio/aws-okta").install buildpath.children
+    cd "src/github.com/segmentio/aws-okta" do
+      system "go", "build", "-ldflags", "-X main.Version=#{version}"
+      bin.install "aws-okta"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/aws-okta version", 0)
```
Comment (fxcoudert) [inline Formula/aws-okta.rb]: We need a test that exercises the some of the functionality of the app.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/aws-okta.rb (hunk 1 above)
Comment (josacar) [reply Formula/aws-okta.rb]: ok, will change

File: Formula/aws-okta.rb (hunk 2 above)
Comment (josacar) [reply Formula/aws-okta.rb]: ups, I forgot it, sorry

**Comment by josacar** [comment]
@fxcoudert how does it looks to you?

**Review by fxcoudert — APPROVED** [review]

**Comment by josacar** [comment]
@fxcoudert Maybe this is asking OSX keychain, I don't know why build is stuck.

If so I'll have to apply this patch:

```diff
diff --git a/Formula/aws-okta.rb b/Formula/aws-okta.rb
index 945b0f13a8..a4edd2d649 100644
--- a/Formula/aws-okta.rb
+++ b/Formula/aws-okta.rb
@@ -18,10 +18,11 @@ class AwsOkta < Formula
   test do
     require 'pty'

-    PTY.spawn("#{bin}/aws-okta add") do |input, output, pid|
+    PTY.spawn("#{bin}/aws-okta --backend file add") do |input, output, pid|
       output.puts "organization\n"; input.gets
       output.puts "username\n"; input.gets
       output.puts "password\n"; input.gets
+      output.puts "\n"; input.gets
       input.gets
       assert_match "Added credentials for user username", input.gets.chomp
       input.close
```

**Comment by josacar** [comment]
@fxcoudert I checked the CI output and the `audit` failed and `test` too because execution expired, I applied the patch to use file backend instead.

**Comment by DomT4** [comment]
This formula needs a `sha256`, heh.

**Comment by josacar** [comment]
@DomT4 Fixed and added also the head version. Thanks!

**Comment by fxcoudert** [comment]
Thanks @josacar for the pull request! This is now merged.

**Pull request closed without merging**